_posts/2024-10-28-using-maven-wildfly-plugin-with-postgresql-ds.adoc

```diff
@@ -0,0 +1,116 @@
+---
+layout: post
+title:  "Using the maven-wildfly-plugin to provision the WildFly to use the PostgreSQL datasource."
+date:   2024-10-28
+tags:   wildfly wildfly-maven-plugin postgresql
+author: liweinan
+description: Introducing the use of the wildfly-maven-plugin to use the PostgreSQL datasource.
+---
+
+Recently I need to configure the WildFly server to use a PostgreSQL datasource for some testings, and I’d like to use the`wildfly-maven-plugin` to automate the server provision process, so I invested some time into the topic and finally make it work. To sum up what I have learned, I have put the usage example in this PR:
+
+* https://github.com/jberet/jberet-examples/pull/8[jberet-examples / add postgresql based repository example #8]
+
+From the above pull request, we can see that the `wildfly-datasources-galleon-pack` feature packfootnote:[https://github.com/wildfly-extras/wildfly-datasources-galleon-pack[wildfly-extras/wildfly-datasources-galleon-pack]] is added into the plugin configuration:
+
+[source,xml]
+----
+<feature-pack>
+    <location>org.wildfly:wildfly-datasources-galleon-pack:3.0.0.Final</location>
+</feature-pack>
+----
+
+And this layer is used:
+
+[source,xml]
+----
+<layers>
+    <layer>postgresql-driver</layer>
+</layers>
+----
+
+The above layer will just add the PostgreSQL driver into the provisioned WildFly server, and it won’t add any datasource by default. If I run the provisioned server with the above feature pack and layer added into the configuration with the following command in the example:
+
+[source,bash]
+----
+$ mvn wildfly:run
+----
+
+I can see from the server log that the PostgreSQL driver is enabled during the server startup process:
+
+[source,txt]
+----
+22:11:00,626 INFO  [org.jboss.as.connector.subsystems.datasources] (ServerService Thread Pool -- 42) WFLYJCA0005: Deploying non-JDBC-compliant driver class org.postgresql.Driver (version 42.5)
+22:11:00,631 INFO  [org.jboss.as.connector.deployers.jdbc] (MSC service thread 1-6) WFLYJCA0018: Started Driver service with driver-name = postgresql
+----
+
+So the feature pack and the layer will prepare the provisioned WildFly server to connect to the PostgreSQL driver, and then we can configure the datasource in the server. In my above example, you can see I use the CLI tool to add the datasource in a script called `setup-server-postgresql.sh`footnote:[https://github.com/jberet/jberet-examples/blob/main/deployment/setup-server-postgresql.sh[jberet-examples/deployment/setup-server-postgresql.sh at main · jberet/jberet-examples]]:
```

Comment:
This can be done without a batch script. I've submitted a PR to update how this can be done in the example https://github.com/jberet/jberet-examples/pull/13.